packages/web-forms/src/components/controls/Input/InputGeopoint.vue

```diff
@@ -0,0 +1,499 @@
+<script setup lang="ts">
+import InputGeopointReadonly from '@/components/controls/Input/InputGeopointReadonly.vue';
+import type { GeopointInputNode, InputValue } from '@getodk/xforms-engine';
+import Button from 'primevue/button';
+import PrimeDialog from 'primevue/dialog';
+import PrimeProgressSpinner from 'primevue/progressspinner';
+import { computed, inject, ref } from 'vue';
+
+interface InputGeopointProps {
+	readonly question: GeopointInputNode;
+}
+
+const props = defineProps<InputGeopointProps>();
+const submitPressed = inject<boolean>('submitPressed');
+const isInvalid = computed(() => props.question.validationState.violation?.valid === false);
+const coords = ref<GeolocationCoordinates | null>(null);
+const showDialog = ref(false);
+const disabled = computed(() => props.question.currentState.readonly === true);
+const geoLocationError = ref<boolean>(false);
+const watchID = ref<number | null>(null);
+const startTime = ref<number | null>(null);
+const elapsedTime = ref(0);
+// TODO: fix TypeScript check so it doesn't take types from NodeJS
+let intervalID: NodeJS.Timeout | undefined;
+
+const value = computed((): InputValue<'geopoint'> => {
+	return props.question.currentState.value;
+});
+
+/**
+ * Default accuracy in meters that can usually be reached by modern devices given enough time.
+ */
+const ACCURACY_THRESHOLD_DEFAULT = 5;
+const accuracyThreshold = computed<number>(() => {
+	return props.question.nodeOptions.accuracyThreshold ?? ACCURACY_THRESHOLD_DEFAULT;
+});
+
+/**
+ * Default unacceptable accuracy in meters, which is about the length of a city block.
+ */
```

Comment:
This is the kind of thing where I _really_ wish the language and type system had a way of expressing units. 🙃